ledger/complete/wal/checkpointer.go

```diff
@@ -293,7 +287,22 @@ func CreateCheckpointWriterForFile(dir, filename string, logger *zerolog.Logger)
 // as for each node, the children have been previously encountered.
 // TODO: evaluate alternatives to CRC32 since checkpoint file is many GB in size.
 // TODO: add concurrency if the performance gains are enough to offset complexity.
-func StoreCheckpoint(writer io.Writer, tries ...*trie.MTrie) error {
+func StoreCheckpoint(dir string, fileName string, logger *zerolog.Logger, tries ...*trie.MTrie) error {
```

Comment:
We need to use named return argument in order to capture and set returned error in `defer`.

```suggestion
func StoreCheckpoint(dir string, fileName string, logger *zerolog.Logger, tries ...*trie.MTrie) (err error) {
```